test/unit/ngsiv2/configuration-api_test.js

```diff
@@ -70,7 +71,7 @@ describe('Configuration API support', function() {
 
     beforeEach(function(done) {
         nock.cleanAll();
-
+        originalResource = config.iota.defaultResource;
```

Comment:
Could you elaborate on this change, pls?

Reply:
This is needed due to the removal of `grunt-mocha-test` to execute the tests. It seems that mocha caches modules and files loaded with `require`.
From https://mochajs.org/api/mocha:
> To run tests multiple times (or to run tests in files that are already in the require cache), make sure to clear them from the cache
There are many other references the Internet and the solution seems to be to delete the cache: https://stackoverflow.com/questions/22975864/how-to-require-same-file-in-mocha-test

In our case, this is a problem for the configuration file since we use always the same in all the tests and it is loaded using:
```
config = require('./config-test.js'),
```

Therefore, the file is actually only loaded in the first test. In the following ones, it is used the cached version. The problem happens when the configuration is modified by one test, as happens in https://github.com/telefonicaid/iotagent-json/blob/master/test/unit/ngsiv2/configuration-api_test.js#87. If this change is not reverted, the next tests will not have the appropriate value in ` config.iota.defaultResource`.

`grunt-mocha-test` may take care of this kind of things, probably in https://github.com/pghalliday/grunt-mocha-test/blob/master/tasks/lib/MochaWrapper.js#36.

Nevertheless, you can check that tests fail for the master branch if you use directly mocha:

```
node_modules/mocha/bin/_mocha tools/mocha-globals.js test/unit/ngsiv2/configuration-api_test.js test/unit/ngsiv2/HTTP_reveice_measures-test.js --exit
```

Reply:
Thanks for the explation. If I'm understanding correctly, removing grunt-mocha makes a little more complated developing the test, as each test needs to have care of levaing the config object as it is found at origin.

However, I think is a reasonable price to pay. Thus, NTC